src/settings/settingsdialog.cpp

```diff
@@ -669,3 +707,30 @@ void SettingsDialog::loadSettings()
     }
     ui->shortcutsTreeView->model()->loadShortcuts(settings);
 }
+
+void SettingsDialog::detectTestTextLanguage(QOnlineTranslator *translator, QOnlineTranslator::Engine engine)
```

Comment:
Should be a reference even if we passing a pointer to it (just dereference it). This indicates that we are not expecting `nullpt` here. Same for the helper below.

```suggestion
void SettingsDialog::detectTestTextLanguage(QOnlineTranslator &translator, QOnlineTranslator::Engine engine)
```